### PR TITLE
Feature order content

### PR DESCRIPTION
**Pull request by Shirosaya25**
Don't really know how much should be added at the same time, so trying a merge.

- Added the feature: Ability to rearrange content within a module.
- This required an update to the backend to include a new column in an existing table and an update to the models affected as well.
- Added Angular Material to the project and switched out the base card class for mat-card
- Added methods to persist content priority in frontend.
- Fixed / refactored tests that were failing or now irrelevant

- Notes: This will probably break the current application because the backend needs to be updated as well in order for it to work properly. I believe that displaying modules should still work, but anything other than that that directly modifies or adds content to a module will most likely be broken until the database is updated

### CODE REVIEW

File: src/main.ts
Comment (carolynrehm) [inline src/main.ts:1]: Why is this included in the main.ts file?

File: src/index.html
Comment (carolynrehm) [inline src/index.html:25]: These should be included with npm and included in the styles array rather than in the index.html

Comment (Shirosaya25) [reply src/index.html:25]: I think that was auto generated with including Angular Material. Do you have any idea on how to include it with npm?

File: src/main.ts
Comment (Shirosaya25) [reply src/main.ts:1]: Loaded by Angular Material since that's one of their dependencies

File: src/index.html
Comment (carolynrehm) [reply src/index.html:25]: Okay, if this was generated we don't need to worry about it

File: src/app/components/content-creator-page/content-creator-page.component.ts
Comment (carolynrehm) [inline src/app/components/content-creator-page/content-creator-page.component.ts]: remove unnecessary comments and console logs

Comment (Shirosaya25) [reply src/app/components/content-creator-page/content-creator-page.component.ts]: Should have removed all of the console.logs from the ts files in latest commit

File: src/app/components/module-index-page/module-index-page.component.ts
Comment (carolynrehm) [inline src/app/components/module-index-page/module-index-page.component.ts]: Is a map necessary here? Can multiple Modules be visible at once?

File: src/app/components/module-index-page/module-index-page.component.ts
Comment (carolynrehm) [inline src/app/components/module-index-page/module-index-page.component.ts]: Why use the document object here rather than databinding?

File: src/app/components/module-index-page/module-index-page.component.ts
Comment (Shirosaya25) [reply src/app/components/module-index-page/module-index-page.component.ts]: Made the map into a variable

File: src/app/components/module-index-page/module-index-page.component.ts
Comment (Shirosaya25) [reply src/app/components/module-index-page/module-index-page.component.ts]: Artifact of last batch that worked on it. Removed and attached ngModel in the HTML

File: src/app/components/module-index-page/module-index-page.component.ts
Comment (carolynrehm) [reply src/app/components/module-index-page/module-index-page.component.ts]: What then is the difference between the active module and the contentVisible and selModule? They seem to all be referencing the same thing

Comment (carolynrehm) [reply src/app/components/module-index-page/module-index-page.component.ts]: +contentActive

Comment (Shirosaya25) [reply src/app/components/module-index-page/module-index-page.component.ts]: Apparently nothing. I squashed contentVisible and selModule and changed the references to them to activeModule

Comment (Shirosaya25) [reply src/app/components/module-index-page/module-index-page.component.ts]: contentActive is the mapping of what piece of content is considered active in each module. That should be a map since it tracks which piece is active even if you close that module

File: src/app/models/Module.ts
Comment (carolynrehm) [inline src/app/models/Module.ts]: please remove this

Comment (Shirosaya25) [reply src/app/models/Module.ts]: Removed

File: src/app/pipes/content-filter.pipe.ts
Comment (carolynrehm) [inline src/app/pipes/content-filter.pipe.ts]: Where is this filter used?

Comment (Shirosaya25) [reply src/app/pipes/content-filter.pipe.ts]: Feature I was going to implement but deemed to complicated. Deleted in the next commit

**Review by carolynrehm — CHANGES_REQUESTED** [review]
Is there a reason why a pipe needs to be used to order content? Angular generally [advises against it](https://angular.io/guide/pipes#appendix-no-filterpipe-or-orderbypipe).  It seems like we only need an initial ordering, so we could just provide a sorted array of lists to begin with.

Furthermore, sorting and filtering operations like these, ideally, should be performed on the back end. If these cannot be changed, it might be better to rename/reorganize the service. It's hard to know what "UtilService" would provide

**Comment by Shirosaya25** [comment]
I tried removing the pipe for ordering content and everything seems to work correctly.

I also renamed util to sortSearch and replaced the refs where needed

File: src/app/components/module-index-page/module-index-page.component.html
Comment (carolynrehm) [inline src/app/components/module-index-page/module-index-page.component.html]: Why add this explicitly?

Comment (Shirosaya25) [reply src/app/components/module-index-page/module-index-page.component.html]: forgot to remove it when testing to make sure it functioned without the pipe

**Review by carolynrehm — APPROVED** [review]
Looks good! I would wait until the back end branch is ready to merge though

**Comment by carolynrehm** [comment]
This pull requests should not include content ordering. Merge that code into dev or remove the unrelated implementation from this pull request

**Comment by carolynrehm** [comment]
@Shirosaya25 a commit has been merged into dev, pull these updates into this branch and it will be merged

**Comment by Shirosaya25** [comment]
@carolynrehm I pulled the changes from dev. Just waiting for Travis to finish

**Pull request merged**
